.github/workflows/publish-prerelease.yml

```diff
@@ -103,6 +103,44 @@ jobs:
         env:
           NPM_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
 
+      - name: Publish Pre-Release version to NPM
+        id: publish_pre_release_version
+        run: |
+          version=$(node -p 'require("./lerna.json").version')
+          echo "::set-output name=LIB_VERSION::${version}"
+          echo "Publishing DEV version - v${version}-alpha.${{ github.run_id }}"
+          make move-package-json-to-dist
+          make prerelease-version-upgrade-${version}-alpha.${{ github.run_id }}
+          make pretty
+          git add .
+          git commit -m "chore: prelrease version"
+          make publish-packages-next
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_PUBLISH_TOKEN }}
+
+      - name: Trigger E2E test workflow
+        uses: actions/github-script@v5
+        env:
+          PR_REF: '${{ steps.cpr.outputs.pull-request-head-sha }}'
+          PR_VERSION: '${{ steps.publish_pre_release_version.outputs.LIB_VERSION }}-alpha.${{ github.run_id }}'
+        with:
+          github-token: ${{ secrets.GH_REPOSITORY_ADMIN_TOKEN }}
+          script: |
+            const sha = process.env.PR_REF;
```

Comment:
Why are we doing the dispatch like this via code?

Reply:
I just used the official github action script code, I need the PR sha so I can prevent from merging before the "private workflow" finished